solidity/dashboard/src/components/AddTopUpModal.jsx

```diff
@@ -1,39 +1,209 @@
-import React from "react"
+import React, { useState } from "react"
 import AmountForm from "./AmountForm"
 import AddressShortcut from "./AddressShortcut"
+import Tag from "./Tag"
+import * as Icons from "./Icons"
+import { withConfirmationModal } from "./ConfirmationModal"
+import Button from "./Button"
+import Divider from "./Divider"
+import TokenAmount from "./TokenAmount"
+import { KEEP } from "../utils/token.utils"
+import { add } from "../utils/arithmetics.utils"
+import { useModal } from "../hooks/useModal"
 
 const AddTopUpModal = ({
   authorizerAddress,
   beneficiary,
   operatorAddress,
-  ...formProps
+  currentAmount,
+  onBtnClick: submitConfirmationModal,
+  onCancel,
+  availableAmount,
+}) => {
+  const [step, setStep] = useState(1)
+  const [amount, setAmount] = useState("0")
+
+  const onSubmit = (values) => {
+    if (step === 1) {
+      setAmount(values.amount)
+      setStep((preveStep) => preveStep + 1)
+    } else if (step === 2) {
+      submitConfirmationModal({ amount })
+    }
+  }
+
+  const amountInTokenUnit = KEEP.fromTokenUnit(amount)
```

Comment:
Bad variable name. We change the `amount` from Token Unit, so it's not in token unit after